userproject/include/spikeRecorder.h

```diff
@@ -26,11 +26,14 @@ class SpikeWriterText
         }
     }
 
+// GCC 4.x does not provide a move constructor for ofstream
+#if defined(__GNUC__) && (__GNUC__ > 4)
```

Comment:
I'm afraid a more complex test is need to deal with Visual C++ and clang (which for extra fun also defines ``__GNUC__``):

```c++
!defined(__GNUC__) ||     __clang__ ||  __GNUC__ > 4
```
maybe